test/Translate/Ops/assign.p4

```diff
@@ -2,29 +2,226 @@
 
 // NOTE: Assertions have been autogenerated by utils/generate-test-checks.py
 
+// CHECK: #[[$ATTR_0:.+]] = #p4hir.int<1> : !b10i
+// CHECK: #[[$ATTR_1:.+]] = #p4hir.int<1> : !infint
+// CHECK: #[[$ATTR_2:.+]] = #p4hir.int<2> : !b10i
+
+// CHECK-LABEL:   p4hir.func action @assign
 action assign() {
+    // CHECK:         %[[VAL_0:.*]] = p4hir.variable ["res"] : <!b10i>
+    // CHECK:         %[[VAL_1:.*]] = p4hir.const #int1_b10i
+    // CHECK:         %[[VAL_2:.*]] = p4hir.cast(%[[VAL_1]] : !b10i) : !b10i
+    // CHECK:         %[[VAL_3:.*]] = p4hir.variable ["lhs", init] : <!b10i>
+    // CHECK:         p4hir.assign %[[VAL_2]], %[[VAL_3]] : <!b10i>
+    // CHECK:         %[[VAL_4:.*]] = p4hir.const #int2_b10i
+    // CHECK:         %[[VAL_5:.*]] = p4hir.cast(%[[VAL_4]] : !b10i) : !b10i
+    // CHECK:         %[[VAL_6:.*]] = p4hir.variable ["rhs", init] : <!b10i>
+    // CHECK:         p4hir.assign %[[VAL_5]], %[[VAL_6]] : <!b10i>
     bit<10> res;
 
     bit<10> lhs = 1;
     bit<10> rhs = 2;
 
+    // CHECK:         %[[VAL_7:.*]] = p4hir.read %[[VAL_3]] : <!b10i>
+    // CHECK:         p4hir.assign %[[VAL_7]], %[[VAL_0]] : <!b10i>
     res = lhs;
+
+    // CHECK:         %[[VAL_8:.*]] = p4hir.read %[[VAL_3]] : <!b10i>
+    // CHECK:         %[[VAL_9:.*]] = p4hir.read %[[VAL_6]] : <!b10i>
+    // CHECK:         %[[VAL_10:.*]] = p4hir.binop(add, %[[VAL_8]], %[[VAL_9]]) : !b10i
+    // CHECK:         p4hir.assign %[[VAL_10]], %[[VAL_0]] : <!b10i>
     res = lhs + rhs;
 }
 
-// CHECK-LABEL:   p4hir.func action @assign()
-// CHECK:         %[[VAL_0:.*]] = p4hir.variable ["res"] : <!b10i>
-// CHECK:         %[[VAL_1:.*]] = p4hir.const #int1_b10i
-// CHECK:         %[[VAL_2:.*]] = p4hir.cast(%[[VAL_1]] : !b10i) : !b10i
-// CHECK:         %[[VAL_3:.*]] = p4hir.variable ["lhs", init] : <!b10i>
-// CHECK:         p4hir.assign %[[VAL_2]], %[[VAL_3]] : <!b10i>
-// CHECK:         %[[VAL_4:.*]] = p4hir.const #int2_b10i
-// CHECK:         %[[VAL_5:.*]] = p4hir.cast(%[[VAL_4]] : !b10i) : !b10i
-// CHECK:         %[[VAL_6:.*]] = p4hir.variable ["rhs", init] : <!b10i>
-// CHECK:         p4hir.assign %[[VAL_5]], %[[VAL_6]] : <!b10i>
-// CHECK:         %[[VAL_7:.*]] = p4hir.read %[[VAL_3]] : <!b10i>
-// CHECK:         p4hir.assign %[[VAL_7]], %[[VAL_0]] : <!b10i>
-// CHECK:         %[[VAL_8:.*]] = p4hir.read %[[VAL_3]] : <!b10i>
-// CHECK:         %[[VAL_9:.*]] = p4hir.read %[[VAL_6]] : <!b10i>
-// CHECK:         %[[VAL_10:.*]] = p4hir.binop(add, %[[VAL_8]], %[[VAL_9]]) : !b10i
-// CHECK:         p4hir.assign %[[VAL_10]], %[[VAL_0]] : <!b10i>
+// CHECK-LABEL:   p4hir.func action @op_assign_ref
+action op_assign_ref() {
+    // CHECK:           %[[VAL_0:.*]] = p4hir.const #[[$ATTR_0]]
+    // CHECK:           %[[VAL_1:.*]] = p4hir.cast(%[[VAL_0]] : !b10i) : !b10i
+    // CHECK:           %[[VAL_2:.*]] = p4hir.variable ["lhs", init] : <!b10i>
```

Comment:
Just grab variables here. And use RHS / LHS as substitution names for the sake of clarity.